src/main/java/com/cleancommunity/posting/PostingFactory.java

```diff
@@ -3,36 +3,29 @@
 public class PostingFactory
 {
     private int numInstancesMade;
+    private PostingDAO postingDAO;
 
-    PostingFactory() {
+    public PostingFactory() {
 
-        // Query database to get number of postings
+        // TODO Query database to get number of postings instead
         this.numInstancesMade = 0;
+        this.postingDAO = new MysqlPostingDAO();
 
     }
 
     public Posting createPostingByRequest(String request) {
 
+        int postingId = extractPostingIdFromRequest(request);
+        Posting posting = postingDAO.getPostingById(postingId);
+
         numInstancesMade++;
-        Posting posting = new Posting(request, numInstancesMade);
 
         return posting;
     }
 
-    // public Posting createPostingById(int id) {
-    //
-    //     Posting posting = new Posting(id);
-    //     numInstancesMade++;
-    //
-    //     return new Posting(id);
-    // }
-
-    // Why is this private?
-    private Posting getPostingFromDatabase(int id) {
-
-        // Querying from database will be next milestone
-
-        return null;
+    private int extractPostingIdFromRequest(String request) {
```

Comment:
Should we perhaps just have a RequestParser class? Where all parsing methods for client requests exist?

Reply:
I think so, this was the primary thing I was referencing when I said I did some unsavory things in the name of progress.